dm/relay/relay_test.go

```diff
@@ -453,14 +455,44 @@ func (t *testRelaySuite) TestHandleEvent(c *C) {
 		c.Assert(errors.Cause(handleErr), Equals, reader2.err)
 	}
 
+	// reader return fake rotate event
+	reader2.err = nil
+	reader2.result.Event = fakeRotateEv
+	// writer return error to force handleEvents return
+	writer2.err = errors.New("writer error for testing")
+	// return with the annotated writer error
+	_, err = r.handleEvents(ctx, reader2, transformer2, writer2)
+	c.Assert(errors.Cause(err), Equals, writer2.err)
+	// after handle rotate event, we save and flush the meta immediately
+	c.Assert(r.meta.Dirty(), Equals, false)
+	{
+		lm := r.meta.(*LocalMeta)
+		c.Assert(lm.BinLogName, Equals, "mysql-bin.666888")
+		c.Assert(lm.BinLogPos, Equals, uint32(1234))
+		filename := filepath.Join(lm.baseDir, lm.currentUUID, utils.MetaFilename)
+		lm2 := &LocalMeta{}
+		_, err := toml.DecodeFile(filename, lm2)
+		c.Assert(err, IsNil)
+		c.Assert(lm2.BinLogName, Equals, "mysql-bin.666888")
+		c.Assert(lm2.BinLogPos, Equals, uint32(1234))
+	}
+	{
+		lm := r.meta.(*LocalMeta)
+		backupUuid := lm.currentUUID
+		lm.currentUUID = "not exist"
+		_, err = r.handleEvents(ctx, reader2, transformer2, writer2)
+		c.Assert(os.IsNotExist(errors.Cause(err)), Equals, true)
+		lm.currentUUID = backupUuid
+	}
+
 	// reader return valid event
 	reader2.err = nil
 	reader2.result.Event = rotateEv
 
 	// writer return error
 	writer2.err = errors.New("writer error for testing")
 	// return with the annotated writer error
-	_, err = r.handleEvents(ctx, reader2, transformer2, writer2)
+	_, err = r.handleEvents(context.Background(), reader2, transformer2, writer2)
```

Comment:
why here need to use backgroud ctx?

Reply:
it's unstable to use a timeout context which'll expire in 10ms and this case is not expecting to test it, it causes this case fail sometimes.